content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,256 @@
+
+```[js]
+addEventListener('fetch', event => {
+    event.respondWith(handleRequest(event.request))
+})
+
+async function handleRequest(request) {
+    try {
+        const body = await request.text();   // Get the body from the response
+        const data = JSON.parse(body);       // Parse body to JavaScript object
+
+        if (!data) {
+            return jsonResponse(null, 'Request Body cannot be empty!');
+        }
+
+        /* Ably specific code here in the sample */
+
+        return new Response ({ text: "I'm ok!" });
+
+    } catch (error) {
+        return new Response(error, { status: 500 });
+    }
+}
+
+```
+
+h2. How the game works
+
+The game is a React app, created using "Create-React-App":https://github.com/facebook/create-react-app, that uses the react-router-dom npm package for routing.
+It's made up of two top level React components, one for the game and the other for an admin dashboard.
+
+The player uses the arrow keys on their keyboard to drive a post van around five pre-set maps (which can be seen in the [MAPS DIRECTORY]). The objective of the game is to deliver a parcel to a postbox on each level. Every time a parcel is delivered, we're publishing a message to an Ably channel and a new postbox is shown to the user.
+
+The Ably Reactor will listen for these publish events and call our Cloudflare Worker with the body of the published message. Once our Cloudflare Worker receives the message from the Ably Reactor, it will publish its own message to the Ably Channel. The dashboard is subscribed to these messages and will display the numnber of moves it took for the player to complete each level. In a real world scenario this might be the number of deliveries left on our driver's shift.
+
+<a href="/images/tutorials/reactor/reactor-event-cloudflare-sequence-diagram.png" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-event-cloudflare-sequence-diagram.png" style="width: 100%" alt="Webapp sequence diagram">
+</a>
+
+h2. Configuring the game
+
+The first thing we need to do to be able to use the provided code is to create a key.js file in the *./src/* directory. Inside this file you'll export your API key as a variable:
+
+```[js]
+export const ABLY_API_KEY = "YOUR_API_KEY"
+```
+
+Add your own Ably API key inbetween the quotation marks!
+
+You will also need to add the API key into the *./cloudflare-worker/index.js* file. Replace the variable near the top of the file.
+
+h2. Configuring Cloudflare
+
+Configure your Cloudflare Worker as per "these instructions in the Cloudflare Docs":https://developers.cloudflare.com/workers/quickstart/#configure.
+
+During configuration of your Cloudflare worker, you're going edit the *./cloudflare-worker/wrangler.toml* file to include your Cloudflare account_id. (You might be wondering where your Cloudflare API key goes - in order to deploy a Worker using Wrangler, you'll configure Wrangler using it's config subcommand so your API keys never touch the Worker code itself.)
```

Comment:
We reference files that should exist somewhere in a Git repo I assume. Can we link to those please?